src/iterator.rs

```diff
@@ -33,13 +34,13 @@ impl Default for GoIter {
 }
 
 impl Iterator for GoIter {
-    type Item = StdResult<KV>;
+    type Item = FfiResult<KV>;
 
     fn next(&mut self) -> Option<Self::Item> {
         let next_db = match self.vtable.next_db {
             Some(f) => f,
             // TODO: return None here???
-            None => return Some(Err(generic_err("iterator vtable not set"))),
+            None => return Some(Err(FfiError::Other)),
```

Comment:
I think we can remove the comment above, right? Seems to be clearly an error.

Having an error strong on `FfiError::Other` is probably nice for debugging. The information from before ("iterator vtable not set") is lost now

Reply:
See comment above

Reply:
Oh, i forgot to respond to the first part here. I do not think that we should be returning `None` here, so yeah i'll remove the comment